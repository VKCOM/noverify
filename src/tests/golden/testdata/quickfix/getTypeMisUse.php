<?php

function getTypeMisUse(mixed $var) {
  if (gettype($var) === "string") {
  }

  if (gettype($var) == "double") {
  }

  if (gettype($var) !== "array") {
  }

  if (gettype($var) != "boolean") {
  }

  if (gettype($var) === "object" && true) {
  }

  if (gettype(getTypeMisUse($var)) === "integer") {
  }

  if (gettype(getTypeMisUse($var)) != "resource") {
  }
}
