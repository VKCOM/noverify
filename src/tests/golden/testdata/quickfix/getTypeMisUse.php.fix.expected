<?php

function getTypeMisUse(mixed $var) {
  if (is_string($var)) {
  }

  if (is_float($var)) {
  }

  if (!is_array($var)) {
  }

  if (!is_bool($var)) {
  }

  if (is_object($var) && true) {
  }

  if (is_int(getTypeMisUse($var))) {
  }

  if (!is_resource(getTypeMisUse($var))) {
  }
}
